src/inference/include/openvino/runtime/properties.hpp

```diff
@@ -0,0 +1,663 @@
+    constexpr static const auto mutability = mutability_;  //!< Property readability
+
+    /**
+     * @brief Constructs property access variable
+     * @param str_ property name
+     */
+    constexpr BaseProperty(const char* name_) : _name{name_} {}
+
+    /**
+     * @brief return property name
+     * @return Pointer to const string key representation
+     */
+    const char* name() const {
+        return _name;
+    }
+
+    /**
+     * @brief compares property name
+     * @return true if string is the same
+     */
+    bool operator==(const std::string& str) const {
+        return _name == str;
+    }
+
+    /**
+     * @brief compares property name
+     * @return true if string is the same
+     */
+    friend bool operator==(const std::string& str, const BaseProperty<T, mutability_>& property) {
+        return property == str;
+    }
+
+private:
+    const char* _name = nullptr;
+};
+template <typename T, PropertyMutability M>
+inline std::ostream& operator<<(std::ostream& os, const BaseProperty<T, M>& property) {
+    return os << property.name();
+}
+}  // namespace util
+/** @endcond */
+
+/**
+ * @brief This class is used to bind property name with value type
+ * @tparam T type of value used to set or get property
+ */
+template <typename T, PropertyMutability mutability_ = PropertyMutability::RW>
+struct Property : public util::BaseProperty<T, mutability_> {
+    using util::BaseProperty<T, mutability_>::BaseProperty;
+    /**
+     * @brief Constructs property
+     * @tparam Args property constructor arguments types
+     * @param args property constructor arguments
+     * @return Pair of name and type erased value.
+     */
+    template <typename... Args>
+    inline std::pair<std::string, Any> operator()(Args&&... args) const {
+        return {this->name(), Any::make<T>(std::forward<Args>(args)...)};
+    }
+};
+
+/**
+ * @brief This class is used to bind read-only property name with value type
+ * @tparam T type of value used to pass or get property
+ */
+template <typename T>
+struct Property<T, PropertyMutability::RO> : public util::BaseProperty<T, PropertyMutability::RO> {
+    using util::BaseProperty<T, PropertyMutability::RO>::BaseProperty;
+};
+
+/**
+ * @brief Read-only property to get a std::vector<PropertyName> of supported read-only properies.
+ *
+ * This can be used as a compiled model property as well.
+ *
+ */
+static constexpr Property<std::vector<PropertyName>, PropertyMutability::RO> supported_properties{
+    "SUPPORTED_PROPERTIES"};
+
+/**
+ * @brief Read-only property to get a std::vector<std::string> of available device IDs
+ */
+static constexpr Property<std::vector<std::string>, PropertyMutability::RO> available_devices{"AVAILABLE_DEVICES"};
+
+/**
+ * @brief Read-only property which defines support of import/export functionality by plugin
+ */
+static constexpr Property<bool, PropertyMutability::RO> import_export_support{"IMPORT_EXPORT_SUPPORT"};
```

Comment:
do we need this property then? it's one of the device capabilities now
CC @nosovmik see below

Reply:
removed